src/stream/KeyExchange.js

```diff
@@ -506,19 +514,20 @@ export function SubscriberKeyExchange(client, { groupKeys = {} } = {}) {
 
     const next = Scaffold([
         async () => {
-            [sub] = await Promise.all([
-                SubscriberKeyExhangeSubscription(client, getGroupKeyStore, encryptionUtil),
-                encryptionUtil.onReady(),
-            ])
+            return encryptionUtil.onReady()
+        },
+        async () => {
+            sub = await SubscriberKeyExhangeSubscription(client, getGroupKeyStore, encryptionUtil)
             return async () => {
                 mem.clear(getGroupKeyStore)
                 if (!sub) { return }
                 const cancelTask = sub.cancel()
                 sub = undefined
-                await cancelTask()
+                await cancelTask
```

Comment:
Should this same fix be applied to line 288, where we currently do `await cancelTask()`?

Reply:
Fixed in cb576fdc